lambda/business-wallet/batch-transactions/smart-contracts/contracts/Voting.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity ^0.8.0;
+
+contract Voting {
+
+    struct proposal {
+        uint256 id;
+        string name;
+        uint256 voteCount;
+        string description;
+    }
+  
+  
+    mapping(address => bool) public voters;
+
+    mapping(address => uint256) public stakes;
+
+    mapping(uint256 => proposal) public proposals;
+
+    uint256 public proposalCount;
+
+    constructor() {
+        proposalCount = 0;
+    }
+
+    function addProposal(string memory name, string memory description) public {
+        proposalCount++;
+        proposals[proposalCount] = proposal(proposalCount, name, 0, description);
+    }
+
+    function stake() public payable {
+        require(msg.value >= 100000000000000, "You need to stake some ETH");
+        stakes[msg.sender] = msg.value;
+    }
+
+    function unstake(address user) public {
+        require(voters[user] == false, "You have already voted");
+        require(stakes[user] == 100000000000000, "You need to stake some ETH");
+        stakes[user] = 0;
+    }
+
+    function vote(address voter, uint256 proposalId) public {
```

Comment:
This will allow anyone to vote for anyone

Reply:
yeah thats fine, voter can vote for any proposal as per their own choice.

Reply:
fixed